osxcollector/output_filters/summary_filters/html.py

```diff
@@ -98,10 +121,62 @@ def end_of_lines(self):
             .val {
                 color: #00ff00;
             }
+
+            dl, dt {
+                margin-bottom: 10px;
+                margin-top: 10px;
+            }
+
         </style></head><body>''')
         self._print_header('Very Readable Output Bot')
         self._print_para('Let\'s see what\'s up with this machine.')
 
+        if (self._group_by_iocs):
+            self.summarize_by_ioc()
+        else:
+            self.summarize_by_threat_indicator()
+
+        self._print_para('Very Readable Output Bot')
+        self._print_para('#kaythanksbye')
+
+        self._write('</body></html>')
+
+        return []
+
+    def summarize_by_ioc(self):
+
+        if self._group_key:
+            self._print_header('Table of contents (sorted by {0})'.format(self._group_key), level=2)
+            self._write('<ul id="toc">')
+            for ioc_key, ioc_value in self._iocs_by_key.iteritems():
+                self._print_section_link(
+                    ioc_key, ioc_key, len(ioc_value))
+            self._print_section_link('iocs', 'Remaining IOCs not tagged by key', len(self._iocs))
+            self._write('</ul>')
+
+        for ioc_key, ioc_value in self._iocs_by_key.iteritems():
```

Comment:
As self._iocs_by_key is only set when self._group_key is set, this should be indented as the above clause.